### PR TITLE
add `fillInputData` option argument to `formatTransaction` on `call` function in rpc_method_wrappers

### LINKED ISSUE #6355: Data is missing from methods.myMethod
needs investigation, from discord user. Looks like the change from data/input has affected methods.mymethod:
Is there a way to force the format of the jsonRPC calls made by methods.myMethod to include the right fields?

Currently generates (4.1.0rc)

```
{"jsonrpc":"2.0","id":"b4c23746-97b2-4512-9dc2-2492d81151ce","method":"eth_call","params":[{"to":"0x5FbDB2315678afecb367f032d93F642f64180aa3","input":"0x70a082310000000000000000000000007a06dcfb82c84eb0e94bec11161012a3374ad98f"},"latest"]}

```

4.0.3

```
{"jsonrpc":"2.0","id":"b4c23746-97b2-4512-9dc2-2492d81151ce","method":"eth_call","params":[{"to":"0x5FbDB2315678afecb367f032d93F642f64180aa3","input":"0x70a082310000000000000000000000007a06dcfb82c84eb0e94bec11161012a3374ad98f","data":"0x70a082310000000000000000000000007a06dcfb82c84eb0e94bec11161012a3374ad98f"},"latest"]}

```

When it should be generating

```
{"jsonrpc":"2.0","id":"b4c23746-97b2-4512-9dc2-2492d81151ce","method":"eth_call","params":[{"to":"0x5FbDB2315678afecb367f032d93F642f64180aa3","data":"0x70a082310000000000000000000000007a06dcfb82c84eb0e94bec11161012a3374ad98f"},"latest"]}

```

Using either option currently gets met with:
```
{
    "jsonrpc": "2.0",
    "id": "b4c23746-97b2-4512-9dc2-2492d81151ce",
    "error": {
        "code": -32602,
        "message": "unknown field `input`, expected one of `from`, `to`, `gasPrice`, `maxFeePerGas`, `maxPriorityFeePerGas`, `gas`, `value`, `data`, `nonce`, `chainId`, `accessList`, `type`"
    }
}
```

#### Steps to reproduce the behavior

1. [First step]
2. [Second step]
3. [and so on...]

### PR DESCRIPTION
**Pull request by moshmage**
## Description

The `fillInputData` option on `formatTransaction` call that was introduced [with this change](https://github.com/web3/web3.js/pull/6294/files#diff-0c52e30746e2e9181931331752f8cb162bc84dba5e7a8cd093509681c8c2707fR64) left `eth_call` method unchanged. In my case, `4.1.0 + hardhat` was unable to call a simple `ERC20.name()` returning `unkown function selector` (even if the contract had a fallback).

Adding this argument to the function call when `call()` is made fixed it on my side.

- (should) Fix #6355

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by luu-alex** [comment]
Hey there @moshmage . As well I have a [pr](https://github.com/web3/web3.js/pull/6377) that addresses different issues related to that change, but i'll merge this branch with mines. Thanks a ton for opening this PR and if there are any other issues you are facing, feel free to open another issue/pr

**Review by avkos — APPROVED** [review]

**Comment by luu-alex** [comment]
Hey there @moshmage  do you have an code snippet you can share of how your using this code? it'll help me out a ton :)

**Comment by moshmage** [comment]
> Hey there @moshmage do you have an code snippet you can share of how your using this code? it'll help me out a ton :)

sure @luu-alex -- head over to https://github.com/moshmage/dappkit/blob/3.x.x/test/models/base-model.spec.ts#L59 (where it fails the test): the `await model.contract.methods.name().call()` is calling the web3js implementation (`.contract.methods = new web3.eth.Contract(abi, address).methods`). This will only fail on a hardhat local chain (I tested it against a ganache fork of ethereum and it went smoothly). you can check the contract here https://github.com/moshmage/dappkit/blob/3.x.x/contracts/token/ERC20/ERC20.sol#L41 (excuse the mess, in the process of cleaning up ^^')

You can read how to set-up a dev-env on https://github.com/moshmage/dappkit/blob/3.x.x/how-to/contributing/readme.md

ps: The way I got to the "fill input data was" that I just echoed out the json params that `web3js@1.x.x` was doing against the current version and noticed that the `1.x.x` call had data and input, and not just one - after adding the `fillInputData: true` to a local-fork of web3js the tests went by smoothly as well.

**Comment by luu-alex** [comment]
Thanks alot this is really helpful! 

We are actually trying to get away from using the default call to be both `data` and `input`, we made the default to be only filling `input` in the `eth_call` and I think the error you are running into is due to hardhat could only be accepting `data`. 

I'll be adding to the PR so that you can use a flag when creating your contract to include in your method calls to either send  `data`, `input` or both. Does this solution work for you?

**Comment by moshmage** [comment]
> Thanks alot this is really helpful!
> 
> We are actually trying to get away from using the default call to be both `data` and `input`, we made the default to be only filling `input` in the `eth_call` and I think the error you are running into is due to hardhat could only be accepting `data`.
> 
> I'll be adding to the PR so that you can use a flag when creating your contract to include in your method calls to either send `data`, `input` or both. Does this solution work for you?

Being fair to hardhat, the only place where `input` is documented is on `geth execution-apis` documentation repo [here](https://github.com/ethereum/execution-apis/blob/c4089414bbbe975bbc4bf1ccf0a3d31f76feb3e1/src/schemas/transaction.yaml#L301) -- everywhere else, `input` is not even mentioned (including [eth.org oficial docs](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_call))

That said, as long as web3js plays ball with hardhat it'll be cool by me;

But I'm afraid that this change will add a "we will deal with any and all edge-cases" precedent and I don't wanna be the cause for it :P

Wouldn't it be better if we go to the [`handleRevert`](https://docs.web3js.org/api/web3-core/class/Web3Config/#handleRevert) route and have it on the Web3Config, instead of "per contract" basis -- that way userland can simply turn the option on and be done with it - no need to understand the underlyings of the Contract class ^^

**Comment by luu-alex** [comment]
I understand, i'm not too sure why the geth documentation and eth official docs are different but we have been trying hard to follow the geth repo as much as possible.

I think your suggestion is great! It will be included as a web3config.
